reactor-core/src/main/java/reactor/core/publisher/DelegateProcessor.java

```diff
@@ -99,10 +99,11 @@ public int getBufferSize() {
 		                .scanOrDefault(Attr.CAPACITY, super.getBufferSize());
 	}
 
+	// super.getError() returns null by default and this method would throw NPE in that
+	// case, however the parent's contract is to return nullable Throwable
 	@Override
-	@Nullable
```

Comment:
Have you tried explicit `@NonNull` instead of `@SuppressWarnings`?

Reply:
The problem is actually in the argument passed to `scanOrDefault` which is required to be non-null, otherwise the method throws. Somehow the contract of that method is that it should not accept null values, yet there is extra verification using `Object.requireNonNull` in place in the body. This is the only weird use of it that can pass on a null argument. Since the class is deprecated I don't want to spend much more time investigating.